pkg/api/dashboard.go

```diff
@@ -332,7 +332,7 @@ func (hs *HTTPServer) PostDashboard(c *models.ReqContext, cmd models.SaveDashboa
 	}
 
 	dashSvc := dashboards.NewService(hs.SQLStore)
```

Comment:
nit: You could do something like `	ctx = context.WithValue(ctx, "unified_alerting_enabled", hs.Cfg.UnifiedAlerting.Enabled)` and avoid passing down the whole struct. The string can be a type or a cost just like you've done with `// CtxCfg is used as a key to get configuration from the context
type CtxCfg struct{}`

Reply:
I had actually done it that way originally but it doesn't pass the linter:

```
[lint-backend:2] pkg/api/dashboard.go:335:65: SA1029: should not use built-in type string as key for value; define your own type to avoid collisions (staticcheck)
[lint-backend:3] 	dashboard, err := dashSvc.SaveDashboard(context.WithValue(ctx, "unified_alerting_enabled", hs.Cfg.UnifiedAlerting.Enabled), dashItem, allowUiUpdate)
```

Reply:
Ah, that makes sense - now I get the reasoning behind the type.

Reply:
But I bet I could set a bool in the CtxCfg struct and only pass that so that we're not copying the entire config. Thanks!